File: Frontend/app/lib/models/report_model.dart
class Report {
  final String id;
  final String description;
  final double latitude;
  final double longitude;
  final String imageHash;

  Report({required this.id, required this.description, required this.latitude, required this.longitude, required this.imageHash});

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'] ?? '',
      description: json['description'] ?? '',
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      imageHash: json['imageHash'] ?? '',
    );
  }
}

File: Geoportal.App/Models/ReportModel.cs
public class ReportModel
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string DeviceId { get; set; } = string.Empty;
}